apps/dapp-console/app/settings/components/ClaimRebateDialog.tsx

```diff
@@ -0,0 +1,156 @@
+import Image from 'next/image'
+
+import {
+  Button,
+  Dialog,
+  DialogContent,
+  DialogHeader,
+  DialogTrigger,
+} from '@eth-optimism/ui-components'
+import { Text } from '@eth-optimism/ui-components/src/components/ui/text/text'
+import { Address, Transaction, formatEther } from 'viem'
+import { useCallback, useMemo, useState } from 'react'
+import { MAX_CLAIMABLE_AMOUNT } from '@/app/constants/rebate'
+import { shortenAddress } from '@eth-optimism/op-app'
+import Link from 'next/link'
+
+export type ClaimRebateDialogProps = {
+  contractAddress: Address
+  contractDeploymentTransaction: Transaction
+  amountClaimed?: bigint
+  isVerified?: boolean
+  children: React.ReactNode
+}
+
+export const ClaimRebateDialog = ({
+  contractAddress,
+  contractDeploymentTransaction,
+  amountClaimed = BigInt(0),
+  isVerified = false,
+  children,
+}: ClaimRebateDialogProps) => {
+  const [isOpen, setOpen] = useState(false)
+
+  const gasFeeToReimburst = useMemo(() => {
```

Comment:
backend will figure out the real final value, the FE will also do the calculation so it doesn't need to make an extra backend call.